### PR TITLE
Do not spawn mobs too close to player

### PR DESCRIPTION
**Pull request by LogicParrot**
Mobs shouldn't spawn in a 24-block radius around the player. This commit takes care of that.

### CODE REVIEW

**Comment by LogicParrot** [comment]
I used `cWorld::FindClosestPlayer`. Is it safe? a comment in its code warns about it.

File: src/Mobs/Creeper.cpp [hunk 1]
```diff
@@ -27,7 +27,7 @@ void cCreeper::Tick(std::chrono::milliseconds a_Dt, cChunk & a_Chunk)
 {
 	super::Tick(a_Dt, a_Chunk);
 
-	if (!TargetIsInRange() && !m_BurnedWithFlintAndSteel)
+	if ((m_Target == nullptr) || (!TargetIsInRange() && !m_BurnedWithFlintAndSteel))
```
Comment (LogicParrot) [inline src/Mobs/Creeper.cpp]: stray code

**Comment by madmaxoft** [comment]
You should rebase onto current master (and work in the main repo instead of your own)

**Comment by LogicParrot** [comment]
Rebased.

**Comment by LogicParrot** [comment]
Should we have a policy regarding tabs? Different IDEs treat them differently and we often git mass tab changes when different devs work on one file.

**Comment by madmaxoft** [comment]
Historically, we kept tabs on otherwise empty lines, because the then-used IDE of choice, MSVC2008, made it so. Nowadays we prefer no tabs on empty lines in new code, but don't want to touch old code just for this change, so we prefer to keep the formatting as-is, but accept PRs that change the formatting in files they touch.

**Comment by madmaxoft** [comment]
The `FindClosestPlayer()` has a threading problem potential - as soon as it returns a `cPlayer` object, you cannot know if it hasn't been freed in the meantime from another thread.
Since you're only using it to check *whether* there is a player, you're on the safe side.

There is a plan to rewrite the function into a callback-based `DoWithClosestPlayer()`, it should be trivial to convert your code when it comes to this.

**Comment by LogicParrot** [comment]
We have a problem. This neutralizes all egg spawns because they are too close to a player.

**Comment by SphinxC0re** [comment]
So, this only applies for passive-aggressive and aggressive mobs?

**Comment by LogicParrot** [comment]
@sphinxc0re I believe it applies to all mobs (it broke passive breeding at some point and then we fixed it)

**Pull request merged**